tests/test_cq.py

```diff
@@ -131,7 +132,9 @@ def test_create_cq_ex(self):
                 try:
                     with CQEX(ctx, cq_init_attrs_ex):
                         pass
-                except PyverbsError as ex:
+                except PyverbsRDMAError as ex:
+                    if 'not implemented' in str(ex):
+                        raise unittest.SkipTest('Create extended CQ is not supported')
```

Comment:
I think it would be better to add the ibv_cq_init_attr_ex - isn't it possible that a device supports creation of extended CQ but some attributes are not supported? (same for all of this patch)

Reply:
But this patch handles the case where the provider doesn't implement the create_cq_ex() at all.

Reply:
I'm not asking for a complex logic. No point having a patch that only works properly for devices that do not support *at all* and wait for someone else to make it helpful to partial support.

Reply:
Could you please add more info about how ibv_cq_init_attr_ex can help in partial support?

I'm thinking about extending the "if" statement to check for "not supported" to handle the partial support.

Reply:
ibv_create_qp_ex is the same as ibv_create_qp if a minimual set of attributes are requested. A provider can support or not any combination of the attributes.

Reply:
Well, there are two failure cases here, The first one is that the provider doesn't implement the verb at all (This patch handle it), and the second one is that provider implements the verb partially, In this case, the only way to determine the error is to check errno value (I'll update the patch to check for "not supported").

@jgunthorpe and @EdwardSro : Thoughts?